### PR TITLE
cherry-pick PR from 1.0

### PR DESCRIPTION
**Pull request by critical27**
cherry-pick storage related PR from Mar 27th to Oct 1st.
Rely on https://github.com/vesoft-inc/nebula-common/pull/255, need it merge at first.
This is a nightmare...

### CODE REVIEW

File: src/storage/StorageFlags.h [hunk 1]
```diff
@@ -33,4 +33,8 @@ DECLARE_int32(reader_handlers);
 
 DECLARE_bool(enable_multi_versions);
 
+DECLARE_bool(enable_multi_versions);
```
Comment (panda-sheep) [inline src/storage/StorageFlags.h:36]: Repeated

File: src/storage/StorageServer.cpp [hunk 2]
```diff
@@ -162,7 +166,11 @@ bool StorageServer::start() {
             storageServer_->setStopWorkersOnStopListening(false);
             storageServer_->setInterface(std::move(handler));
 
-            storageSvcStatus_.store(STATUS_RUNNING);
+            ServiceStatus expected = STATUS_UNINITIALIZED;
+            if (!storageSvcStatus_.compare_exchange_strong(expected, STATUS_RUNNING)) {
+                LOG(ERROR) << "Impossible! How could it happens!";
```
Comment (panda-sheep) [inline src/storage/StorageServer.cpp:171]: happen

File: src/storage/StorageServer.cpp [hunk 3]
```diff
@@ -185,7 +193,11 @@ bool StorageServer::start() {
             adminServer_->setStopWorkersOnStopListening(false);
             adminServer_->setInterface(std::move(handler));
 
-            adminSvcStatus_.store(STATUS_RUNNING);
+            ServiceStatus expected = STATUS_UNINITIALIZED;
+            if (!adminSvcStatus_.compare_exchange_strong(expected, STATUS_RUNNING)) {
+                LOG(ERROR) << "Impossible! How could it happens!";
```
Comment (panda-sheep) [inline src/storage/StorageServer.cpp:198]: ditto

**Review by panda-sheep — COMMENTED** [review]
A huge work, excellent！

File: src/storage/StorageFlags.h (hunk 1 above)
Comment (critical27) [reply src/storage/StorageFlags.h]: good catch, fixed!

File: src/storage/StorageServer.cpp (hunk 2 above)
Comment (critical27) [reply src/storage/StorageServer.cpp]: Fixed, thx

File: src/storage/StorageServer.cpp (hunk 3 above)
Comment (critical27) [reply src/storage/StorageServer.cpp]: Fixed, thx

**Review by panda-sheep — APPROVED** [review]
Impressive！

**Review by bright-starry-sky — APPROVED** [review]
LGTM.

**Pull request merged**